tuf/api/metadata.py

```diff
@@ -485,22 +523,35 @@ def __init__(
         spec_version: str,
         expires: datetime,
         meta: Mapping[str, Any],
+        unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ) -> None:
         super().__init__(_type, version, spec_version, expires)
         # TODO: Add class for meta
         self.meta = meta
+        self.unrecognized_fields = unrecognized_fields
 
     @classmethod
     def from_dict(cls, timestamp_dict: Mapping[str, Any]) -> "Timestamp":
         """Creates Timestamp object from its dict representation. """
         common_args = cls._common_fields_from_dict(timestamp_dict)
         meta = timestamp_dict.pop("meta")
-        return cls(*common_args, meta)
+        # Store unregonized fields. Read ADR 0008.
+        expected_fields = [
+            "_type",
+            "version",
+            "spec_version",
+            "expires",
+            "meta",
+        ]
+        unrecognized_fields = _get_unrecognized_fields(
+            timestamp_dict, expected_fields
+        )
+        return cls(*common_args, meta, unrecognized_fields)
 
     def to_dict(self) -> Dict[str, Any]:
         """Returns the dict representation of self. """
         timestamp_dict = self._common_fields_to_dict()
-        timestamp_dict.update({"meta": self.meta})
+        timestamp_dict.update({"meta": self.meta, **self.unrecognized_fields})
```

Comment:
I think this will fail if you create Timestamp using the constructor with unrecognized_fields unset. Maybe the instance variable should not be Optional (set to {} if argument is None)

Reply:
should we add a test for pure constructor as well (as in not using `from_dict()`)?